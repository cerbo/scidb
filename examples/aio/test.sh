#!/bin/bash

rm -rf test.out

iquery -aq "
project(
 apply(
  op_count(list('instances')), 
  test, iif(count<3, 
             string(throw('This test needs more instances to run')), 
             'Instance Count OK')
 ), 
 test
)" > test.out 2>&1

iquery -anq "remove(zero_to_255)"                > /dev/null 2>&1
iquery -anq "remove(minus_128_to_127)"           > /dev/null 2>&1
iquery -anq "remove(zero_to_65536)"              > /dev/null 2>&1
iquery -anq "remove(minus_32768_to_32767)"       > /dev/null 2>&1
iquery -anq "remove(big_n_wild)"                 > /dev/null 2>&1

iquery -anq "store( build( <val:string> [x=0:65535999,100000,0],  string(x % 256) ), zero_to_255 )"                       > /dev/null 2>&1
iquery -anq "store( build( <val:string> [x=0:65535999,100000,0],  ' ' + string(x % 256 - 128) + ' ' ), minus_128_to_127 )"            > /dev/null 2>&1
iquery -anq "store( build( <val:string> [x=0:65535999,100000,0],  string(x % 65536) + ' ' ), zero_to_65536 )"                   > /dev/null 2>&1
iquery -anq "store( build( <val:string> [x=0:65535999,100000,0],  ' ' + string(x % 65536 - 32768) ), minus_32768_to_32767 )"    > /dev/null 2>&1
iquery -anq "
store(
 build( <val:string null> [x=0:59999999, 100000, 0],
   iif( x % 6 = 0, string(x / 6 * 100) + ' ',
   iif( x % 6 = 1, ' ' + string(uint64(x) * 200000000000),
   iif( x % 6 = 2, '   ' + string(double(x) / 123.4) + ' ', 
   iif( x % 6 = 3, string(x) + '_abc',
   iif( x % 6 = 4, iif( x % 4 = 0, ' true', 'FalSe '),
                   string(null))))))
 ),
 big_n_wild
)"                                                                                                                        > /dev/null 2>&1

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('127',  int8(null)))"                                           >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('128',  int8(null)))"                                           >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-128', int8(null)))"                                           >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-129', int8(null)))"                                           >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('255', uint8(null)))"                                           >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('256', uint8(null)))"                                           >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('0',   uint8(null)))"                                           >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-1',  uint8(null)))"                                           >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('32767',  int16(null)))"                                        >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('32768',  int16(null)))"                                        >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-32768', int16(null)))"                                        >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-32769', int16(null)))"                                        >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('65535',  uint16(null)))"                                       >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('65536',  uint16(null)))"                                       >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('0',      uint16(null)))"                                       >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-1',     uint16(null)))"                                       >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('2147483647', int32(null)))"                                    >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('2147483648', int32(null)))"                                    >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-2147483648', int32(null)))"                                   >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-2147483649', int32(null)))"                                   >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('4294967295', uint32(null)))"                                   >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('4294967296', uint32(null)))"                                   >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('0',          uint32(null)))"                                   >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-1',         uint32(null)))"                                   >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('4294967295', uint32(null)))"                                   >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('4294967296', uint32(null)))"                                   >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('0',          uint32(null)))"                                   >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-1',         uint32(null)))"                                   >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('9223372036854775807',  int64(null)))"                          >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('9223372036854775808',  int64(null)))"                          >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-9223372036854775808', int64(null)))"                          >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-9223372036854775809', int64(null)))"                          >> test.out

iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('18446744073709551615', uint64(null)))"                         >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('18446744073709551616', uint64(null)))"                         >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('0', uint64(null)))"                                            >> test.out
iquery -aq "build(<val:double null> [x=0:0,1,0], dcast('-1', uint64(null)))"                                           >> test.out

time iquery -aq "aggregate(apply(zero_to_255, v2, dcast(val, uint8(null))), avg(v2), count(v2), count(*))"             >> test.out
time iquery -aq "aggregate(apply(zero_to_255, v2, dcast(val, int8(null))), avg(v2), count(v2), count(*))"              >> test.out
time iquery -aq "aggregate(apply(minus_128_to_127, v2, dcast(val, uint8(null))), avg(v2), count(v2), count(*))"        >> test.out
time iquery -aq "aggregate(apply(minus_128_to_127, v2, dcast(val, int8(null))), avg(v2), count(v2), count(*))"         >> test.out

time iquery -aq "aggregate(apply(zero_to_65536, v2, dcast(val, uint16(null))), avg(v2), count(v2), count(*))"          >> test.out
time iquery -aq "aggregate(apply(zero_to_65536, v2, dcast(val, int16(null))), avg(v2), count(v2), count(*))"           >> test.out
time iquery -aq "aggregate(apply(minus_32768_to_32767, v2, dcast(val, uint16(null))), avg(v2), count(v2), count(*))"   >> test.out
time iquery -aq "aggregate(apply(minus_32768_to_32767, v2, dcast(val, int16(null))), avg(v2), count(v2), count(*))"    >> test.out

time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, uint8(null))), avg(v2), count(v2), count(*))"              >> test.out   
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, int8(null))), avg(v2), count(v2), count(*))"               >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, int16(null))), avg(v2), count(v2), count(*))"              >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, uint16(null))), avg(v2), count(v2), count(*))"             >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, int32(null))), avg(v2), count(v2), count(*))"              >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, uint32(null))), avg(v2), count(v2), count(*))"             >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, int64(null))), avg(v2), count(v2), count(*))"              >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, uint64(null))), avg(v2), count(v2), count(*))"             >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, float(null))), avg(v2), count(v2), count(*))"              >> test.out
time iquery -aq "aggregate(apply(big_n_wild, v2, dcast(val, double(null))), avg(v2), count(v2), count(*))"             >> test.out

time iquery -aq "aggregate(apply(filter(apply(big_n_wild, v2, dcast(val, bool(null))), v2 is not null), v3, iif(v2, 1, 0)), sum(v3), count(*))" >> test.out


rm -rf /tmp/load_tools_test
mkdir /tmp/load_tools_test
touch /tmp/load_tools_test/empty_file
touch /tmp/load_tools_test/file1
echo 'col1    col2    col3
"alex"	1	3.5
"b"ob"	2	4.8
jake	4.0
random	3.1	"extra stuff"
bill 	abc	9
alice	4	not_a_number' >> /tmp/load_tools_test/file1
touch /tmp/load_tools_test/file2
echo '"",1
"abc",2
"def",3
null,4
xyz,4.5' >> /tmp/load_tools_test/file2
ln -s /tmp/load_tools_test/file1 /tmp/load_tools_test/symlink1
mkfifo /tmp/load_tools_test/fifo1
mkdir /tmp/load_tools_test/directory

#This is put here to help Jenkins configurations
chmod -R a+rwx /tmp/load_tools_test

iquery -aq "split('/tmp/load_tools_test/file1', 'lines_per_chunk=3')" >> test.out
iquery -aq "split('/tmp/load_tools_test/file1', 'lines_per_chunk=1')" >> test.out
iquery -aq "split('/tmp/load_tools_test/file1')"                      >> test.out
iquery -aq "split(
            'paths=/tmp/load_tools_test/file1;/tmp/load_tools_test/symlink1',
            'instances=1;2',
            'header=1',
            'lines_per_chunk=2'
            )" >> test.out
cat /tmp/load_tools_test/file2 > /tmp/load_tools_test/fifo1 &
iquery -aq "split(
            'paths=/tmp/load_tools_test/file1;/tmp/load_tools_test/symlink1;/tmp/load_tools_test/fifo1',
            'instances=1;2;0',
            'header=1',
            'lines_per_chunk=2'
            )" >> test.out
iquery -aq "split(
            'paths=/tmp/load_tools_test/file1;/tmp/load_tools_test/file2;/tmp/load_tools_test/directory',
            'instances=1;2;0',
            'lines_per_chunk=2', 
            'delimiter=,'
            )" >> test.out

iquery -anq "remove(foo)" > /dev/null 2>&1
iquery -anq "remove(bar)" > /dev/null 2>&1

iquery -anq "store(build(<val:double> [x=1:8000000,1000000,0], random()), foo)" > /dev/null
iquery -anq "save(foo, 'foo.tsv', -1, 'tsv')" > /dev/null
time iquery -anq "store(project(filter(apply(parse(split('paths=foo.tsv', 'instances=-1'), 'num_attributes=1'), v, dcast(a0, double(null))), v is not null), v), bar)" > /dev/null 
iquery -aq "op_count(bar)" >> test.out

iquery -aq "aio_input('/tmp/load_tools_test/file1', 'num_attributes=3')" >> test.out
iquery -aq "sort(aio_input('/tmp/load_tools_test/file1', 'num_attributes=3', 'buffer_size=40'), a0)" >> test.out
iquery -aq "sort(aio_input(
            'paths=/tmp/load_tools_test/file1;/tmp/load_tools_test/symlink1',
            'instances=1;2',
            'header=1',
            'num_attributes=2',
            'buffer_size=31'
            ),a0)" >> test.out
cat /tmp/load_tools_test/file2 > /tmp/load_tools_test/fifo1 &
iquery -aq "aio_input(
            'paths=/tmp/load_tools_test/file1;/tmp/load_tools_test/symlink1;/tmp/load_tools_test/fifo1',
            'instances=1;2;0',
            'header=1',
            'num_attributes=1'
            )" >> test.out
iquery -aq "sort(aio_input(
            'paths=/tmp/load_tools_test/file1;/tmp/load_tools_test/file2;/tmp/load_tools_test/directory',
            'instances=1;2;0',
            'buffer_size=41', 
            'attribute_delimiter=,',
            'num_attributes=3'
            ), a0)" >> test.out

iquery -anq "remove(bar)" > /dev/null 2>&1
time iquery -anq "store(project(filter(apply(aio_input('paths=foo.tsv', 'instances=-1', 'num_attributes=1'), v, dcast(a0, double(null))), v is not null), v), bar)" > /dev/null
iquery -aq "op_count(bar)" >> test.out

iquery -naq "remove(foo)" > /dev/null 2>&1
iquery -naq "store(apply(build(<v1:float null>[i=1:50,10,0], iif(i%2=0,null,i)), v2, double(i/10.4), v3, 'abcdef'), foo)" > /dev/null
iquery -anq "aio_save(sort(foo,v1,v2,v3,25), 'paths=/tmp/load_tools_test/foo;/tmp/load_tools_test/foo2', 'instances=2;0', 'format=tdv', 'cells_per_chunk=3')" >> test.out
iquery -aq "sort(input(foo, '/tmp/load_tools_test/foo',  0, 'tsv'), v2)" >> test.out
iquery -aq "sort(input(foo, '/tmp/load_tools_test/foo2', 0, 'tsv'), v2)" >> test.out
iquery -anq "aio_save(foo, '/tmp/load_tools_test/foo', 'format=(float null, double, string)', 'cells_per_chunk=8')" >> test.out
iquery -aq "sort(input(foo, '/tmp/load_tools_test/foo', 0, '(float null, double, string)'), v2)" >> test.out
iquery -anq "aio_save(between(sort(foo,v1,v2,v3), 0,9), 'paths=/tmp/load_tools_test/foo', 'instances=0', 'format=tdv', 'null_pattern=?%')" >> test.out
cat /tmp/load_tools_test/foo >> test.out
iquery -anq "aio_save(between(sort(foo,v1,v2,v3), 0,9), 'paths=/tmp/load_tools_test/foo', 'instances=0', 'format=tdv', 'null_pattern=NULL CODE %, BRUH')" >> test.out
cat /tmp/load_tools_test/foo >> test.out
iquery -anq "aio_save(between(sort(foo,v1,v2,v3), 0,9), 'paths=/tmp/load_tools_test/foo', 'instances=0', 'format=tdv', 'null_pattern=%, BRUH')" >> test.out
cat /tmp/load_tools_test/foo >> test.out
iquery -anq "aio_save(apply(show(foo), b, bool(true)),  'paths=/tmp/load_tools_test/foo', 'instances=0', 'format=tdv')" >> test.out
cat /tmp/load_tools_test/foo >> test.out
iquery -anq "aio_save(apply(show(foo), b, bool(true)),  'paths=/tmp/load_tools_test/foo', 'instances=1', 'format=tdv')" >> test.out
cat /tmp/load_tools_test/foo >> test.out
iquery -anq "aio_save(apply(show('filter(foo, \'abc\' = \'def\' )', 'afl'), b, bool(true)),  'path=/tmp/load_tools_test/foo', 'instances=1', 'format=csv+')" >> test.out
cat /tmp/load_tools_test/foo >> test.out

diff test.out test.expected
