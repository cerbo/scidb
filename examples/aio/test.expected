{i} test
{0} 'Instance Count OK'
{x} val
{0} 127
{x} val
{0} null
{x} val
{0} -128
{x} val
{0} null
{x} val
{0} 255
{x} val
{0} null
{x} val
{0} 0
{x} val
{0} null
{x} val
{0} 32767
{x} val
{0} null
{x} val
{0} -32768
{x} val
{0} null
{x} val
{0} 65535
{x} val
{0} null
{x} val
{0} 0
{x} val
{0} null
{x} val
{0} 2.14748e+09
{x} val
{0} null
{x} val
{0} -2.14748e+09
{x} val
{0} null
{x} val
{0} 4.29497e+09
{x} val
{0} null
{x} val
{0} 0
{x} val
{0} null
{x} val
{0} 4.29497e+09
{x} val
{0} null
{x} val
{0} 0
{x} val
{0} null
{x} val
{0} 9.22337e+18
{x} val
{0} null
{x} val
{0} -9.22337e+18
{x} val
{0} null
{x} val
{0} 1.84467e+19
{x} val
{0} null
{x} val
{0} 0
{x} val
{0} null
{i} v2_avg,v2_count,count
{0} 127.5,65536000,65536000
{i} v2_avg,v2_count,count
{0} 63.5,32768000,65536000
{i} v2_avg,v2_count,count
{0} 63.5,32768000,65536000
{i} v2_avg,v2_count,count
{0} -0.5,65536000,65536000
{i} v2_avg,v2_count,count
{0} 32767.5,65536000,65536000
{i} v2_avg,v2_count,count
{0} 16383.5,32768000,65536000
{i} v2_avg,v2_count,count
{0} 16383.5,32768000,65536000
{i} v2_avg,v2_count,count
{0} -0.5,65536000,65536000
{i} v2_avg,v2_count,count
{0} 118.182,11,60000000
{i} v2_avg,v2_count,count
{0} 60,6,60000000
{i} v2_avg,v2_count,count
{0} 20651.5,48755,60000000
{i} v2_avg,v2_count,count
{0} 37142.9,115711,60000000
{i} v2_avg,v2_count,count
{0} 2.75938e+08,18128466,60000000
{i} v2_avg,v2_count,count
{0} 2.75938e+08,18128466,60000000
{i} v2_avg,v2_count,count
{0} 6.15636e+17,28128466,60000000
{i} v2_avg,v2_count,count
{0} 1.3731e+18,25814610,60000000
{i} v2_avg,v2_count,count
{0} 5.7723e+17,30000000,60000000
{i} v2_avg,v2_count,count
{0} 5.7723e+17,30000000,60000000
{i} v3_sum,count
{0} 5000000,10000001
{source_instance_id,chunk_no} value
{0,0} 'col1    col2    col3
"alex"	1	3.5
"b"ob"	2	4.8'
{0,1} 'jake	4.0
random	3.1	"extra stuff"
bill 	abc	9'
{0,2} 'alice	4	not_a_number'
{source_instance_id,chunk_no} value
{0,0} 'col1    col2    col3'
{0,1} '"alex"	1	3.5'
{0,2} '"b"ob"	2	4.8'
{0,3} 'jake	4.0'
{0,4} 'random	3.1	"extra stuff"'
{0,5} 'bill 	abc	9'
{0,6} 'alice	4	not_a_number'
{source_instance_id,chunk_no} value
{0,0} 'col1    col2    col3
"alex"	1	3.5
"b"ob"	2	4.8
jake	4.0
random	3.1	"extra stuff"
bill 	abc	9
alice	4	not_a_number'
{source_instance_id,chunk_no} value
{1,0} '"alex"	1	3.5
"b"ob"	2	4.8'
{1,1} 'jake	4.0
random	3.1	"extra stuff"'
{1,2} 'bill 	abc	9
alice	4	not_a_number'
{2,0} '"alex"	1	3.5
"b"ob"	2	4.8'
{2,1} 'jake	4.0
random	3.1	"extra stuff"'
{2,2} 'bill 	abc	9
alice	4	not_a_number'
{source_instance_id,chunk_no} value
{0,0} '"abc",2
"def",3'
{0,1} 'null,4
xyz,4.5'
{1,0} '"alex"	1	3.5
"b"ob"	2	4.8'
{1,1} 'jake	4.0
random	3.1	"extra stuff"'
{1,2} 'bill 	abc	9
alice	4	not_a_number'
{2,0} '"alex"	1	3.5
"b"ob"	2	4.8'
{2,1} 'jake	4.0
random	3.1	"extra stuff"'
{2,2} 'bill 	abc	9
alice	4	not_a_number'
{source_instance_id,chunk_no} value
{1,0} 'col1    col2    col3
"alex"	1	3.5
"b"ob"	2	4.8
jake	4.0
random	3.1	"extra stuff"
bill 	abc	9
alice	4	not_a_number
'
{2,0} '"",1
"abc"'
{2,1} '2
"def",3
null'
{2,2} '4
xyz,4.5
'
{i} count
{0} 8000000
{tuple_no,dst_instance_id,src_instance_id} a0,a1,a2,error
{0,0,0} 'col1    col2    col3',null,null,'short'
{1,0,0} '"alex"','1','3.5',null
{2,0,0} '"b"ob"','2','4.8',null
{3,0,0} 'jake','4.0',null,'short'
{4,0,0} 'random','3.1','"extra stuff"',null
{5,0,0} 'bill ','abc','9',null
{6,0,0} 'alice','4','not_a_number',null
{n} a0,a1,a2,error
{0} '"alex"','1','3.5',null
{1} '"b"ob"','2','4.8',null
{2} 'alice','4','not_a_number',null
{3} 'bill ','abc','9',null
{4} 'col1    col2    col3',null,null,'short'
{5} 'jake','4.0',null,'short'
{6} 'random','3.1','"extra stuff"',null
{n} a0,a1,error
{0} '"alex"','1','long	3.5'
{1} '"alex"','1','long	3.5'
{2} '"b"ob"','2','long	4.8'
{3} '"b"ob"','2','long	4.8'
{4} 'alice','4','long	not_a_number'
{5} 'alice','4','long	not_a_number'
{6} 'bill ','abc','long	9'
{7} 'bill ','abc','long	9'
{8} 'jake','4.0',null
{9} 'jake','4.0',null
{10} 'random','3.1','long	"extra stuff"'
{11} 'random','3.1','long	"extra stuff"'
{tuple_no,dst_instance_id,src_instance_id} a0,error
{0,0,0} '"abc",2',null
{1,0,0} '"def",3',null
{2,0,0} 'null,4',null
{3,0,0} 'xyz,4.5',null
{0,1,1} '"alex"','long	1	3.5'
{1,1,1} '"b"ob"','long	2	4.8'
{2,1,1} 'jake','long	4.0'
{3,1,1} 'random','long	3.1	"extra stuff"'
{4,1,1} 'bill ','long	abc	9'
{5,1,1} 'alice','long	4	not_a_number'
{0,2,2} '"alex"','long	1	3.5'
{1,2,2} '"b"ob"','long	2	4.8'
{2,2,2} 'jake','long	4.0'
{3,2,2} 'random','long	3.1	"extra stuff"'
{4,2,2} 'bill ','long	abc	9'
{5,2,2} 'alice','long	4	not_a_number'
{n} a0,a1,a2,error
{0} '""','1',null,'short'
{1} '"abc"','2',null,'short'
{2} '"alex"	1	3.5',null,null,'short'
{3} '"b"ob"	2	4.8',null,null,'short'
{4} '"def"','3',null,'short'
{5} 'alice	4	not_a_number',null,null,'short'
{6} 'bill 	abc	9',null,null,'short'
{7} 'col1    col2    col3',null,null,'short'
{8} 'jake	4.0',null,null,'short'
{9} 'null','4',null,'short'
{10} 'random	3.1	"extra stuff"',null,null,'short'
{11} 'xyz','4.5',null,'short'
{i} count
{0} 8000000
Query was executed successfully
{n} v1,v2,v3
{0} 1,0.0961538,'abcdef'
{1} 3,0.288462,'abcdef'
{2} 5,0.480769,'abcdef'
{3} null,0.769231,'abcdef'
{4} null,0.961538,'abcdef'
{5} null,1.15385,'abcdef'
{6} 13,1.25,'abcdef'
{7} 15,1.44231,'abcdef'
{8} 17,1.63462,'abcdef'
{9} null,1.92308,'abcdef'
{10} null,2.11538,'abcdef'
{11} null,2.30769,'abcdef'
{12} 25,2.40385,'abcdef'
{13} 27,2.59615,'abcdef'
{14} 29,2.78846,'abcdef'
{15} null,3.07692,'abcdef'
{16} null,3.26923,'abcdef'
{17} null,3.46154,'abcdef'
{18} 37,3.55769,'abcdef'
{19} 39,3.75,'abcdef'
{20} 41,3.94231,'abcdef'
{21} null,4.23077,'abcdef'
{22} null,4.42308,'abcdef'
{23} null,4.61538,'abcdef'
{24} 49,4.71154,'abcdef'
{n} v1,v2,v3
{0} null,0.192308,'abcdef'
{1} null,0.384615,'abcdef'
{2} null,0.576923,'abcdef'
{3} 7,0.673077,'abcdef'
{4} 9,0.865385,'abcdef'
{5} 11,1.05769,'abcdef'
{6} null,1.34615,'abcdef'
{7} null,1.53846,'abcdef'
{8} null,1.73077,'abcdef'
{9} 19,1.82692,'abcdef'
{10} 21,2.01923,'abcdef'
{11} 23,2.21154,'abcdef'
{12} null,2.5,'abcdef'
{13} null,2.69231,'abcdef'
{14} null,2.88462,'abcdef'
{15} 31,2.98077,'abcdef'
{16} 33,3.17308,'abcdef'
{17} 35,3.36538,'abcdef'
{18} null,3.65385,'abcdef'
{19} null,3.84615,'abcdef'
{20} null,4.03846,'abcdef'
{21} 43,4.13462,'abcdef'
{22} 45,4.32692,'abcdef'
{23} 47,4.51923,'abcdef'
{24} null,4.80769,'abcdef'
Query was executed successfully
{n} v1,v2,v3
{0} 1,0.0961538,'abcdef'
{1} null,0.192308,'abcdef'
{2} 3,0.288462,'abcdef'
{3} null,0.384615,'abcdef'
{4} 5,0.480769,'abcdef'
{5} null,0.576923,'abcdef'
{6} 7,0.673077,'abcdef'
{7} null,0.769231,'abcdef'
{8} 9,0.865385,'abcdef'
{9} null,0.961538,'abcdef'
{10} 11,1.05769,'abcdef'
{11} null,1.15385,'abcdef'
{12} 13,1.25,'abcdef'
{13} null,1.34615,'abcdef'
{14} 15,1.44231,'abcdef'
{15} null,1.53846,'abcdef'
{16} 17,1.63462,'abcdef'
{17} null,1.73077,'abcdef'
{18} 19,1.82692,'abcdef'
{19} null,1.92308,'abcdef'
{20} 21,2.01923,'abcdef'
{21} null,2.11538,'abcdef'
{22} 23,2.21154,'abcdef'
{23} null,2.30769,'abcdef'
{24} 25,2.40385,'abcdef'
{25} null,2.5,'abcdef'
{26} 27,2.59615,'abcdef'
{27} null,2.69231,'abcdef'
{28} 29,2.78846,'abcdef'
{29} null,2.88462,'abcdef'
{30} 31,2.98077,'abcdef'
{31} null,3.07692,'abcdef'
{32} 33,3.17308,'abcdef'
{33} null,3.26923,'abcdef'
{34} 35,3.36538,'abcdef'
{35} null,3.46154,'abcdef'
{36} 37,3.55769,'abcdef'
{37} null,3.65385,'abcdef'
{38} 39,3.75,'abcdef'
{39} null,3.84615,'abcdef'
{40} 41,3.94231,'abcdef'
{41} null,4.03846,'abcdef'
{42} 43,4.13462,'abcdef'
{43} null,4.23077,'abcdef'
{44} 45,4.32692,'abcdef'
{45} null,4.42308,'abcdef'
{46} 47,4.51923,'abcdef'
{47} null,4.61538,'abcdef'
{48} 49,4.71154,'abcdef'
{49} null,4.80769,'abcdef'
Query was executed successfully
?0	0.192308	abcdef
?0	0.384615	abcdef
?0	0.576923	abcdef
?0	0.769231	abcdef
?0	0.961538	abcdef
?0	1.15385	abcdef
?0	1.34615	abcdef
?0	1.53846	abcdef
?0	1.73077	abcdef
?0	1.92308	abcdef
Query was executed successfully
NULL CODE 0, BRUH	0.192308	abcdef
NULL CODE 0, BRUH	0.384615	abcdef
NULL CODE 0, BRUH	0.576923	abcdef
NULL CODE 0, BRUH	0.769231	abcdef
NULL CODE 0, BRUH	0.961538	abcdef
NULL CODE 0, BRUH	1.15385	abcdef
NULL CODE 0, BRUH	1.34615	abcdef
NULL CODE 0, BRUH	1.53846	abcdef
NULL CODE 0, BRUH	1.73077	abcdef
NULL CODE 0, BRUH	1.92308	abcdef
Query was executed successfully
0, BRUH	0.192308	abcdef
0, BRUH	0.384615	abcdef
0, BRUH	0.576923	abcdef
0, BRUH	0.769231	abcdef
0, BRUH	0.961538	abcdef
0, BRUH	1.15385	abcdef
0, BRUH	1.34615	abcdef
0, BRUH	1.53846	abcdef
0, BRUH	1.73077	abcdef
0, BRUH	1.92308	abcdef
Query was executed successfully
foo<v1:float,v2:double NOT NULL,v3:string NOT NULL> [i=1:50,10,0]	true
Query was executed successfully
foo<v1:float,v2:double NOT NULL,v3:string NOT NULL> [i=1:50,10,0]	true
Query was executed successfully
i,schema,b
0,'foo@1<v1:float,v2:double NOT NULL,v3:string NOT NULL> [i=1:50,10,0]',true
